### PR TITLE
feat: Add lock and hide inspector components

### PR DESCRIPTION
**Pull request by cyaiox**
This PR adds the `Lock` and `Hide` components to let users `lock` and `hide` entities in the inspector.

https://github.com/decentraland/js-sdk-toolchain/assets/3170051/55d70d77-3795-4e78-a0db-af33ada882fb

Closes: https://github.com/decentraland/sdk/issues/1067

### CODE REVIEW

File: packages/@dcl/inspector/src/components/Tree/Tree.tsx [hunk 1]
```diff
@@ -212,13 +268,32 @@ export function Tree<T>() {
...
+                      <LockIcon className="lock-icon" size={16} onClick={handleToggleLockComponent} />
+                    ) : (
+                      <UnlockIcon className="unlock-icon" size={16} onClick={handleToggleLockComponent} />
+                    )}
+                    {isEntityHidden ? (
+                      <InvisibleIcon className="invisible-icon" size={16} onClick={handleToggleHideComponent} />
+                    ) : (
+                      <VisibleIcon className="visible-icon" size={16} onClick={handleToggleHideComponent} />
+                    )}
+                  </div>
```
Comment (nicoecheza) [inline packages/@dcl/inspector/src/components/Tree/Tree.tsx:295]: maybe we can move some of this to another component? so we avoid adding more complexity to this one, which is already pretty hard to follow along

Comment (cyaiox) [reply packages/@dcl/inspector/src/components/Tree/Tree.tsx:295]: sure! doing it

**Review by nicoecheza — APPROVED** [review]
